frontend/pages/queries/QueryPage/QueryPage.tsx

```diff
@@ -44,35 +44,43 @@ const QueryPage = ({
   const { isGlobalAdmin, isGlobalMaintainer, isAnyTeamMaintainer } = useContext(
     AppContext
   );
-  const { selectedOsqueryTable, setSelectedOsqueryTable } = useContext(
-    QueryContext
-  );
+  const {
+    selectedOsqueryTable,
+    setSelectedOsqueryTable,
+    setLastEditedQueryName,
+    setLastEditedQueryDescription,
+    setLastEditedQueryBody,
+    setLastEditedQueryObserverCanRun,
+  } = useContext(QueryContext);
 
   const [step, setStep] = useState<string>(QUERIES_PAGE_STEPS[1]);
   const [selectedTargets, setSelectedTargets] = useState<ITarget[]>([]);
   const [isLiveQueryRunnable, setIsLiveQueryRunnable] = useState<boolean>(true);
   const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(true);
-  const [
-    isEditorUsingDefaultQuery,
-    setIsEditorUsingDefaultQuery,
-  ] = useState<boolean>(true);
-  const [typedQueryBody, setTypedQueryBody] = useState<string>("");
-
   const [
     showOpenSchemaActionText,
     setShowOpenSchemaActionText,
   ] = useState<boolean>(false);
 
+  // disabled on page load so we can control the number of renders
+  // else it will re-populate the context on occasion
   const {
```

Comment:
Do you want to disable refetchOnWindowFocus? I noticed a lot of API calls when switching between tabs.